src/pages/TokenDetails/index.tsx

```diff
@@ -26,35 +24,28 @@ export default function TokenDetailsPage() {
     [chain, isNative, pageChainId, timePeriod, tokenAddress]
   )
 
-  const [tokenQueryReference, loadTokenQuery] = useQueryLoader<TokenQuery>(tokenQuery)
-  const [priceQueryReference, loadPriceQuery] = useQueryLoader<TokenPriceQuery>(tokenPriceQuery)
-
-  useEffect(() => {
-    loadTokenQuery({ contract })
-    loadPriceQuery({ contract, duration })
-  }, [contract, duration, loadPriceQuery, loadTokenQuery, timePeriod])
+  const { data: tokenQuery, loading: tokenDataLoading } = useTokenQuery({
+    variables: {
+      contract,
+    },
+  })
 
-  const refetchTokenPrices = useCallback(
-    (t: TimePeriod) => {
-      loadPriceQuery({ contract, duration: toHistoryDuration(t) })
-      setTimePeriod(t)
+  const { data: tokenPriceQuery } = useTokenPriceQuery({
+    variables: {
+      contract,
+      duration,
     },
-    [contract, loadPriceQuery, setTimePeriod]
-  )
+  })
 
-  if (!tokenQueryReference) {
-    return <TokenDetailsPageSkeleton />
-  }
+  if (!tokenQuery || tokenDataLoading) return <TokenDetailsPageSkeleton />
```

Comment:
In what case would `tokenQuery && tokenDataLoading`? I'm afraid this case might an the invalid case, where you would not want to show a skeleton because you have data but it is stale.

Reply:
When the time period is changed would we want to show the stale data in this case or would we want to show the skeleton to inform the user that we are updating?

Reply:
skeleton. what about if the data is refreshed - will this ever happen?

Reply:
The way it is currently setup, no. If we setup polling or a manual refresh function then it could happen.

Reply:
In that case though I think there is still value is showing a skeleton state to show to the user that the data is being refreshed